test/src/testPubSub.cpp

```diff
@@ -0,0 +1,267 @@
+/*
+ * Copyright (c) 2024 General Motors GTO LLC
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * SPDX-FileType: SOURCE
+ * SPDX-FileCopyrightText: 2024 General Motors GTO LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <spdlog/spdlog.h>
+#include <up-client-zenoh-cpp/client/upZenohClient.h>
+#include <up-cpp/transport/builder/UAttributesBuilder.h>
+// #include <up-cpp/uri/serializer/LongUriSerializer.h>
+#include <up-cpp/uri/serializer/MicroUriSerializer.h>
+#include <gtest/gtest.h>
+
+#include <condition_variable>
+#include <mutex>
+#include <deque>
+#include <vector>
+#include <tuple>
+#include <chrono>
+#include <random>
+#include <sys/socket.h>
+#include <poll.h>
+#include <unistd.h>
+#include <signal.h>
+
+using namespace uprotocol::utransport;
+using namespace uprotocol::v1;
+using namespace uprotocol::uri;
+using namespace uprotocol::uuid;
+using namespace uprotocol::client;
+
+// capture and queue onReceive() for verification by the sender
+
+template <bool INTERPROC> 
+class MessageCapture : public UListener {
+public:
+    typedef std::vector<std::uint8_t>    Data;
```

Comment:
Minor: `using` style type definitions are more consistent with the rest of the code base.
```suggestion
    using Data = std::vector<std::uint8_t>;
```

Reply:
Will be resolved in next push.